### ISSUE TITLE
No assert for padding check in the checksum utility function

### ISSUE THREAD

**Issue by gnagorsk** [pmem/issues#118]
https://github.com/pmem/nvml/blob/master/src/common/util.c#L454

This if statement can only be possible if the (addr % 4) equals (csump % 4) there should be an assert for this check.

Also the function header is saying that the function assumes little endian:
https://github.com/pmem/nvml/blob/master/src/common/util.c#L437
But then later it is using the le32toh and htole64 functions that are architecture specific for little/big endian. Isn't this actually making the function support both endians?

**Comment by krzycz** [comment]
Right, it's based on the assumption that *addr* and *csump* are always properly aligned. This is true in case of our library, but we can add the asserts to catch the invalid arguments - just in case.
